diff --git a/src/glsl_optimizer_lib.gyp b/src/glsl_optimizer_lib.gyp
index 1094480..a22650a 100644
--- a/src/glsl_optimizer_lib.gyp
+++ b/src/glsl_optimizer_lib.gyp
@@ -110,6 +110,8 @@
         'glsl/lower_vec_index_to_cond_assign.cpp',
         'glsl/lower_vec_index_to_swizzle.cpp',
         'glsl/lower_vector.cpp',
+        'glsl/lower_vector_insert.cpp',
+        'glsl/lower_named_interface_blocks.cpp',
         'glsl/opt_algebraic.cpp',
         'glsl/opt_array_splitting.cpp',
         'glsl/opt_constant_folding.cpp',
@@ -131,6 +133,8 @@
         'glsl/opt_tree_grafting.cpp',
         'glsl/opt_vectorize.cpp',
         'glsl/opt_vector_splitting.cpp',
+        'glsl/opt_flip_matrices.cpp',
+        'glsl/opt_dead_builtin_varyings.cpp',
         'glsl/program.h',
         'glsl/ralloc.c',
         'glsl/ralloc.h',
